docs/spec.md

```diff
@@ -51,6 +51,9 @@
 
 Each JSON message must be an object containing a field called `op` which identifies the type of message.
 
+Additionally, each JSON message must encode any binary data fields (e.g.,
+protobuf or flatbuffer schemas) with base64.
```

Comment:
🤷  Technically, this isn't a requirement of the websocket protocol. Someone could choose to represent binary data with base36 instead of base64. The only requirement of websockets in general is that text messages are valid UTF8, and we're additionally adding a requirement that they should be valid JSON. Maybe it would be more accurate to mention that Foxglove Studio expects the schema field to be base64-encoded if the schemas are represented with binary data, and link to [these cods](https://foxglove.dev/docs/studio/connection/custom)? But I see that below we already link to this on the docs specific to the `schema` field...

Reply:
We should call out the well-known binary schemas by name, so people scanning the docs can quickly see "I'm using protobufs, oh I need to base64-encode my FileDescriptorSet" or "I'm using ROS, I can send my message definitions directly as UTF8".

Reply:
You think that should live in the actual ws-protocol spec, or in an appendix or in studio docs?

Reply:
Thinking through this, maybe we should make base64 encoding of binary data in JSON messages part of the spec. I don't think it harms any use cases (i.e. someone wanting to use base36 encoding instead and expecting Studio or other clients to support that), and it clarifies the implementation details.